discovery/discovery_test.go

```diff
@@ -47,6 +51,30 @@ func StubOrchestrators(addresses []string) []*lpTypes.Transcoder {
 	return orchestrators
 }
 
+func TestDeadLock(t *testing.T) {
+	gmp := runtime.GOMAXPROCS(50)
+	defer runtime.GOMAXPROCS(gmp)
```

Comment:
Do we still need to set GOMAXPROCS?

Reply:
Yes. This test not always fails even with `GOMAXPROCS`, without it it fails with much less probability. (This test can't be made to fail reliably - it depends on go scheduler (or OS?) which decides who take lock first).

Reply:
I'd take a closer look to see if there's another way we can control the scheduling, but my brain is tired right now. LGTM 😄